src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -862,6 +862,41 @@ async def SendCommand(self, nodeid: int, endpoint: int, payload: ClusterObjects.
             interactionTimeoutMs=interactionTimeoutMs, busyWaitMs=busyWaitMs, suppressResponse=suppressResponse).raise_on_error()
         return await future
 
+    async def SendBatchCommands(self, nodeid: int, commands: typing.List[typing.Tuple[int, ClusterObjects.ClusterCommand, typing.Type]],
+                                timedRequestTimeoutMs: typing.Union[None, int] = None,
+                                interactionTimeoutMs: typing.Union[None, int] = None, busyWaitMs: typing.Union[None, int] = None,
+                                suppressResponse: typing.Union[None, bool] = None):
+        '''
+        Send a batch of cluster-object encapsulated commands to a node and get returned a future that can be awaited upon to receive
+        the responses. If a valid responseType is passed in, that will be used to deserialize the object. If not,
+        the type will be automatically deduced from the metadata received over the wire.
+
+        nodeId: Target's Node ID
+        commands: A list of tuples of type (endpoint, cluster-object, response-type):
+        timedWriteTimeoutMs: Timeout for a timed invoke request. Omit or set to 'None' to indicate a non-timed request.
+        interactionTimeoutMs: Overall timeout for the interaction. Omit or set to 'None' to have the SDK automatically compute the
+                              right timeout value based on transport characteristics as well as the responsiveness of the target.
+        busyWaitMs: How long to wait in ms after sending command to device before performing any other operations.
+        suppressResponse: Do not send a response to this action
+
+        Returns:
+            - Array of command respones in the same order as what was given in `commands`. The type of the response is defined by the command.
+        Raises:
+            - InteractionModelError if error with sending of InvokeRequestMessage fails as a whole.
```

Comment:
Can you document here what happens if one command fails, but the rest succeed?

Reply:
Will do, once I push next PR up